src/components/NameFilter.tsx

```diff
@@ -0,0 +1,26 @@
+import { useSearchParams } from 'react-router-dom';
+import { getSearchWith } from '../utils/searchHelper';
+
+export const NameFilter = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const query = searchParams.get('query') || '';
+
+  return (
+    <p className="control has-icons-left">
+      <input
+        type="text"
+        className="input"
+        placeholder="Search"
+        value={query}
+        onChange={event => setSearchParams(
```

Comment:
![image](https://user-images.githubusercontent.com/62388820/184168341-7d0df109-4765-4d48-ab43-35434d769847.png)

Use `debounce`
Whole table re-renders on every letter, because no one will ever see those intermediate results